HIP/hip-1056.md

```diff
@@ -0,0 +1,1769 @@
+we have chosen to perform extra work on consensus nodes to produce clean block streams.
+
+**Designed for Streaming or as Block files:** The stream is designed to both be packaged up as self-contained block
+files and to be streamed as individual items with low latency.
+
+#### Design for Consolidation
+
+The block stream is designed to be a continuous stream of items. This allows for continuous streaming over gRPC. The
+order of items is significant, for example all items between and inclusive of a `BlockHeader` and `BlockProof` represent
+contents of one block. All items following an `EventTransaction` before the next `EventTransaction` are related to that
+first transaction. Block items for one block can also be packaged in a `Block` message so they can be stored in a
+protobuf file and served as a single entity if desired.
+
+![Block Stream Items](../assets/hip-blocks/block-stream-items.svg)
+
+```
+NOTE: Update: Add missing Round header after Block Header to image
+```
+
+The Block Item design promotes granular stream processing ensuring all items in a block prior to the proof can be sent
+out immediately after processing. The result is reduced end-to-end latency for use cases prioritizing speed (e.g.
+high-frequency trading). Additionally, this means a following block will not be sent until a block proof is pushed out
+over the stream.
+
+#### Design for Input & Output streams
+
+The block stream items are split into two types: input and output. The input items are those known after consensus but
+before transaction execution and are the input to transaction execution. Output items are those produced during
+transaction execution. These are hashed separately into two different *block stream* merkle sub-trees. This is done so
+we can separate out a subset of historical blocks with just their input items, or just output items. That subset block stream could then be used
+for transaction replay or transaction execution verification, for example.
+
+![Block Stream Merkle Tree](../assets/hip-blocks/block-stream-merkle.svg)
+
+Consensus nodes have a `state merkle tree`, which is a merkle tree containing the current state of the consensus node.
+The block stream has a separate `block stream merkle tree` which is a merkle tree of all the items in the block stream
+over the entire history of the stream. Of course, older blocks can be represented by their hash rather than the actual
+full contents, so it is possible to have a very large block stream merkle tree *in concept* that is still very efficient
+to use.
+
+#### Design for Verifiability
+
+Existing record files contain the following cryptographic properties.
+
+Record files are streams in a blockchain, with the hash of previous file included in the next record file, ensuring
+immutability. Additionally, each consensus node on the Hedera mainnet signs a hash of the record files as they are
```

Comment:
```suggestion
Record file contents are summarized in a chain of hashes, with each file's contents containing the hash of previous file; so that the hash of the current function depends on the contents of the entire record stream. Additionally, each consensus node on the Hedera mainnet signs the hash of each record file as it is
```

Reply:
...containing the hash of _the_ previous file...